src/transformers/trainer.py

```diff
@@ -2226,16 +2226,25 @@ def _load_best_model(self):
                         state_dict["_smp_is_partial"] = False
                         load_result = model.load_state_dict(state_dict, strict=True)
                 else:
-                    # We load the model state dict on the CPU to avoid an OOM error.
-                    if self.args.save_safetensors and os.path.isfile(best_safe_model_path):
-                        state_dict = safetensors.torch.load_file(best_safe_model_path, device="cpu")
+                    # If train base_8_bit_models using PEFT & LoRA, assume that adapter have been saved properly.
+                    if hasattr(model, "base_model") and getattr(model.base_model, "is_8bit_serializable", False):
+                        if hasattr(model, "active_adapter") and hasattr(model, "load_adapter"):
```

Comment:
Can you add another check that checks if the file `adapter_model.bin` is present before loading it? If it is not the case it should trigger a warning advising users to add the callback that you have mentioned IMO